src/NUnitEngine/nunit.engine.api/IExtensionService.cs

```diff
@@ -58,6 +58,11 @@ public interface IExtensionService
         /// <param name="typeName"></param>
         /// <param name="enabled"></param>
         void EnableExtension(string typeName, bool enabled);
+
+        /// <summary>
+        /// Get an enumeration of all registered extensions of a specific type.
+        /// </summary>
+        IEnumerable<T> GetExtensions<T>();
```

Comment:
I'm adding a new method to the public IExtensionService API here, primarily or the purpose of making MasterTestRunner more testable - however this seems a sensible method to expose in general. If anyone can think of a way this could bite us in future, I'd appreciate the feedback!